adi_update_boot.sh

```diff
@@ -483,16 +550,32 @@ while true
         ;;
       y)
         ### Remove boot partition
-        echo "Removing boot partition files..."
-        rm -rf $FAT_MOUNT/*
+        echo "Removing boot files from /boot..."
+        cd "$FAT_MOUNT" || exit 1
+        rm -rf !(*.dat|*.elf|bootcode.bin|cmdline.txt|config.txt|LICENSE.*)
```

Comment:
things like this are rather dangerous... What if the user has some .dat or .elf files that we are not supposed to manage?

Reply:
The command above deletes everything besides what match regex. So files *.dat, *.elf, bootcode.bin... will remained untouched.

Do you think it would be better to use something like 
`cd /boot & rm -rf zynq-* zynqmp-* versal-* socfpga_* overlays` 
deleting all ADI folders and let files which are directly in /boot untouched?

Reply:
> The command above deletes everything besides what match regex. So files *.dat, *.elf, bootcode.bin... will remained untouched.

Ahh yes, correct...

> Do you think it would be better to use something like
cd /boot & rm -rf zynq-* zynqmp-* versal-* socfpga_* overlays
deleting all ADI folders and let files which are directly in /boot untouched?

Yeah, I think we should somehow keep a list of files that we manage and just delete those. For example, if just we delete all the overlays, we might end up deleting some custom overlays done by users... At the very least we should prompt a message showing what we are going to do and give the decision to the user to proceed or not.

Reply:
> At the very least we should prompt a message showing what we are going to do and give the decision to the user to proceed or not.

We already prompt a message and let the user choose. Now the message is:
`Next step will delete boot files from /boot and will extract the new ones !`
`Are you sure you want to continue? (y/n)`


From my point of view, I would go with first version:
`rm -rf !(*.dat|*.elf|bootcode.bin|cmdline.txt|config.txt|LICENSE.*)`
which delete everything from /boot besides the files that match regex. And which, actually, can be updated a bit:
`rm -rf !(fixup*.dat | start*.elf | bootcode.bin | cmdline.txt | config.txt | LICENCE.broadcom)`

Plus updating message for user, something like:
`WARNING ! Next step will delete everything from /boot. Please move or copy any personal files to rootfs partition before proceeding !!!`

The actual problem is that we need somehow to not delete *.dat, *.elf, bootcode.bin, cmdline.txt and so on from /boot partition whenever we update ADI boot files, since those specific files are generated only by building kuiper.